### ISSUE TITLE
SOCKS requests are having the source IP mangled to localhost

### ISSUE THREAD

**Issue by sertys3** [luminati-io/luminati-proxy#186]
Cheers, team,
this is one wicked bug that i got last week.

TL;DR
If you want Session(IP) per machine to work for you with networked client requests, use HTTP proxies, not SOCKS

 Since i'm using the "Session(IP) per machine" template in the LPM, i was expecting to have different IPs for different source IP requests. This was not the case and i started investigating why.
I'm using LPM in docker(on linux) on 10.30.0.1 internal vpn(with network_mode:host) and my clients are connecting from 10.30.0.0/16 . I was using SOCKS5 proxies at the time.

But my requests were seen in the Logs section as "Sent from 127.0.0.1:XXXXX", instead of "Sent from 10.30.yy.yy:XXXXX". After jumping all the iptables rules to exclude possible masquerading i was seeing that the requests were reaching the node app with src ip 10.30.yy.yy . 
And then being redirected to lo interface already with "127.0.0.1:XXXXX". 

Is there a piece of code that re-writes packages and sends them to a different "socks" listener to handle them there? I'm too busy now to dissect the code now, but a dev will easily pinpoint this and possible merge a fix so it keeps the original source IP or at least saves it for the session.

Keep it up

**Comment by fudali** [comment]
Hi! Thanks for reaching out and pointing the issue. We scheduled this to be solved in the following month
